src/Core/Services/ExecutionHelper.cs

```diff
@@ -319,8 +321,8 @@ private static void SetContextResult(IMiddlewareContext context, JsonDocument? r
                 SupportedHotChocolateTypes.INT_TYPE => ((IntValueNode)value).ToInt32(),
                 SupportedHotChocolateTypes.LONG_TYPE => ((IntValueNode)value).ToInt64(),
                 SupportedHotChocolateTypes.SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
-                SupportedHotChocolateTypes.FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
-                SupportedHotChocolateTypes.DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                SupportedHotChocolateTypes.FLOAT_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDouble() : ((FloatValueNode)value).ToDouble(),
+                SupportedHotChocolateTypes.DECIMAL_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDecimal() : ((FloatValueNode)value).ToDecimal(),
```

Comment:
How does `value` come to be of type `IntValueNode`:
- Does DAB attempt to cast variables to certain types?
- Is the conversion done by HotChocolate during request processing prior to DAB code?

Depending on the answer to the above question: Is this the right place for a fix or should a fix be added earlier in request processing? The fact that `value`'s type doesn't match `argumentSchema.Type.TypeName().Value` seems weird to me.

Reply:
HotChocolate does this conversion giving the type of input value and then at this place we try to do the casting based on the column type.

This is the earliest place where we check and caste the input values from user.

Reply:
Why put this check? If we refer here: https://chillicream.com/docs/hotchocolate/v13/defining-a-schema/scalars#net-scalars, it states that hotchocolate decimal type is equivalent to .net float type. So instead, we just use methods like float.parse() or double.parse() ? (Reference: https://learn.microsoft.com/en-us/dotnet/api/system.double.parse?view=net-8.0)

Reply:
I didn't understand how we would use that here.  the problem was hotchocolate thinking of the value provided by user as IntNodeType, and we were incorrectly casting it before.

Reply:
If we use the parsing functions, we don't need to worry about the casting done by hotchocolate